### PR TITLE
build: Bump shared eslint config deps to latest

### PR DESCRIPTION
**Pull request by tylerbutler**
Bump client and build-tools release groups to use latest (2.1.0) release of our shared eslint config.

The import/no-deprecated rule is now enabled by default, so most of the changes are related to suppressing or disabling that rule in some packages.

The @typescript-eslint/prefer-nullish-coalescing rule identified a few new places that could be changed, so I did that, which saved a handful of bytes in code size.

There were also some other rules that were triggered for unknown reasons:

- @typescript-eslint/no-misused-promises identified a few new places in React code. I ignored them because I don't know how to fix them, or even if they're legitimate.
- @typescript-eslint/prefer-nullish-coalescing needed to be disabled in some projects because it requires strictNullChecks

### CODE REVIEW

File: packages/drivers/odsp-driver/.eslintrc.js [hunk 1]
```diff
@@ -13,5 +13,6 @@ module.exports = {
 		"@typescript-eslint/no-use-before-define": "off",
 		"@typescript-eslint/strict-boolean-expressions": "off",
 		"@rushstack/no-new-null": "error",
+		"import/no-deprecated": "warn", // This package uses the deprecated ShareLinkTypes type. Once we remove that, we can remove this override. It's set as a warning instead of "off" to serve as a reminder.
```
Comment (tylerbutler) [inline packages/drivers/odsp-driver/.eslintrc.js:16]: @jatgarg Do you know if there are plans to update the odsp-driver to use the new types?

File: experimental/dds/attributable-map/src/interfaces.ts [hunk 2]
```diff
@@ -19,7 +19,6 @@ export interface IValueChanged {
 	 * The value that was stored at the key prior to the change.
 	 */
 	// TODO: Use `unknown` instead (breaking change).
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (alexvy86) [inline experimental/dds/attributable-map/src/interfaces.ts:22]: How come these went away? Was the rule already disabled at the package level?

Comment (tylerbutler) [reply experimental/dds/attributable-map/src/interfaces.ts:22]: Good catch; I don't know why these changes were made or why they are considered OK. Seems like a bug in either eslint or the rule itself. I have
reverted these changes.

File: packages/tools/devtools/devtools-browser-extension/package.json [hunk 3]
```diff
@@ -85,7 +85,8 @@
 		"@fluidframework/container-definitions": "workspace:~",
 		"@fluidframework/container-loader": "workspace:~",
 		"@fluidframework/container-runtime-definitions": "workspace:~",
-		"@fluidframework/eslint-config-fluid": "^2.0.0",
+		"@fluidframework/core-interfaces": "workspace:~",
+		"@fluidframework/eslint-config-fluid": "^2.1.0",
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/tools/devtools/devtools-browser-extension/package.json:89]: Good catch; fixed.

File: packages/dds/merge-tree/src/mergeTreeDeltaCallback.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+/* eslint-disable import/no-deprecated */
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/dds/merge-tree/src/mergeTreeDeltaCallback.ts:5]: I must have gotten really lazy. It's not even really a benefit over individual disables in this file. Fixed.

**Review by anthony-murphy — APPROVED** [review]
This is awesome! Can you please create a feature in the backlog and tag it as tech debt to clean these up

**Pull request merged**